apis/python/src/tiledb/vector_search/object_api/object_index.py

```diff
@@ -0,0 +1,390 @@
+        group.meta["object_reader_kwargs"] = encode_class(object_reader.init_kwargs())
+        group.meta["embedding_source_code"] = get_source_code(embedding)
+        group.meta["embedding_class_name"] = embedding.__class__.__name__
+        group.meta["embedding_kwargs"] = encode_class(embedding.init_kwargs())
+
+        embeddings_array_name = storage_formats[index.storage_version][
+            "INPUT_VECTORS_ARRAY_NAME"
+        ]
+        filters = storage_formats[index.storage_version]["DEFAULT_ATTR_FILTERS"]
+
+        create_embeddings_partitioned_array(
+            uri, embeddings_array_name, group, vector_type, filters
+        )
+        object_metadata_array_uri = object_reader.metadata_array_uri()
+        materialize_object_metadata = False
+        if (
+            object_metadata_array_uri is None
+            and object_reader.metadata_attributes() is not None
+        ):
+            metadata_array_name = storage_formats[index.storage_version][
+                "OBJECT_METADATA_ARRAY_NAME"
+            ]
+            object_metadata_array_uri = f"{uri}/{metadata_array_name}"
+            external_ids_dim = tiledb.Dim(
+                name="external_id",
+                domain=(0, np.iinfo(np.dtype("int32")).max),
+                dtype=np.dtype(np.int32),
+            )
+            external_ids_dom = tiledb.Domain(external_ids_dim)
+            schema = tiledb.ArraySchema(
+                domain=external_ids_dom,
+                sparse=True,
+                attrs=object_reader.metadata_attributes(),
+            )
+            tiledb.Array.create(object_metadata_array_uri, schema)
+            group.add(object_metadata_array_uri, name=metadata_array_name)
+            materialize_object_metadata = True
+        object_metadata_external_id_dim = ""
+        if object_metadata_array_uri is not None:
+            with tiledb.open(object_metadata_array_uri, "r") as object_metadata_array:
+                object_metadata_external_id_dim = (
+                    object_metadata_array.schema.domain.dim(0).name
+                )
+            group.meta["object_metadata_array_uri"] = object_metadata_array_uri
+            group.meta[
+                "object_metadata_external_id_dim"
+            ] = object_metadata_external_id_dim
+
+        group.meta["materialize_object_metadata"] = materialize_object_metadata
+        group.close()
+        return ObjectIndex(uri, config, load_embedding=False, **kwargs)
+
+
+def create_embeddings_partitioned_array(
+    group_uri: str, array_name: str, group: tiledb.Group, vector_type: np.dtype, filters
+):
+    embeddings_array_uri = f"{group_uri}/{array_name}"
+    if tiledb.array_exists(embeddings_array_uri):
+        raise ValueError(f"Array exists {embeddings_array_uri}")
+    partition_id_dim = tiledb.Dim(
+        name="partition_id",
+        domain=(0, flat_index.MAX_INT32),
+        tile=1,
+        dtype=np.dtype(np.uint32),
```

Comment:
the dtype is `uint32` but the domain is max int32

Reply:
fixed